### ISSUE TITLE
[Bug]: Missing Read-only state for Radio Tile component in Carbon Design system

### ISSUE THREAD

**Issue by SwatiKat** [carbon-design-system/carbon#15716]
### Package

@carbon/react

### Browser

Chrome, Safari, Firefox, Edge

### Package version

v1.38.0

### React version

v18.2.0

### Description

Our project team needs read-only states of Radio tiles which can give readable affordance to the users which currently our workaround - disable state is not providing for the following usecase: 
![image (41)](https://github.com/carbon-design-system/carbon/assets/159429105/19cdf3f6-f5e6-48cf-b398-c0d17d245b45)

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-tile--radio

### Steps to reproduce

Missing read-only property in the Action controls

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Application/PAL

IBM products

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by laurenmrice** [comment]
Seems that it would be okay to do this as long as the descriptive text is legible. We can put this issue on a future sprint to do some design explorations of what this would look like.

**Comment by chrisdenais** [comment]
I would like to help with this issue
